### DISCUSSION TITLE
Which situation we use this instead of data class?

### DISCUSSION THREAD

**Discussion by nielinjie** [drewhamilton/Poko#388] [General]
Which situation we use this instead of data class? I don't get the point. Why not just use data class?

**Comment by drewhamilton** [comment]
The main reason is to make compatibility easier to maintain when the class is in a public API: this is hard/tedious to do with generated `copy` and `comlonentN` functions, so Poko leaves those out. See https://jakewharton.com/public-api-challenges-in-kotlin/ for more info.

**Reply by nielinjie** [reply]
Cool, thanks.
